pkg/apis/servicecatalog/types.go

```diff
@@ -398,23 +398,30 @@ type ServiceInstance struct {
 	Status ServiceInstanceStatus
 }
 
-// ServiceInstanceSpec represents the desired state of an Instance.
-type ServiceInstanceSpec struct {
-	// ExternalClusterServiceClassName is the human-readable name of the service
-	// as reported by the broker. Note that if the broker changes
+// DesiredPlan defines the user specification for the desired
+// ServicePlan and ServiceClass. Because there are multiple ways to
+// specify the desired Class/Plan, this structure specifies the
+// allowed ways to specify the intent.
+type DesiredPlan struct {
+	// ExternalClusterServiceClassName is the human-readable name of the
+	// service as reported by the broker. Note that if the broker changes
 	// the name of the ClusterServiceClass, it will not be reflected here,
 	// and to see the current name of the ClusterServiceClass, you should
 	// follow the ClusterServiceClassRef below.
 	//
 	// Immutable.
 	ExternalClusterServiceClassName string
```

Comment:
Probably irrelevant to your PR, but I wonder if the field should have been called `ClusterServiceClassExternalName` instead, otherwise it reads as if we had external vs internal classes/plans.

Reply:
Yeah, that was done a while ago. Original idea was to distinguish between k8s resource names and 'external' names. We could probably call it OSBServiceClassName ?

Reply:
My point is that it's not a name of the external thing, but an external name of the thing. Or more specifically, we are referring to a field `ExternalName` of the `ClusterServiceClass` resource. Hence `ClusterServiceClass` + `ExternalName` => `ClusterServiceClassExternalName`.

> We could probably call it OSBServiceClassName ?

I don't like it. The field is referring to a field name of Service Catalog resource, and user doesn't need to be aware that it has something to do with OSB API.

Reply:
Actually, to continue with this line of thinking... the "External" on there could be confusing to people. And end-user will just see a list of serviceclass names and pick one. The presence of the word 'external' is really not relevant to them since where the service is hosted does not matter. As an end-user I'm being asked to provide the "name" of the "ServiceClass", or technically "ClusterServiceClass" - so that's what I would expect the property name to be.  Ditto for ExternalClusterServicePlanName below.

Reply:
I'm sorry I missed your replies here to my reply. Let's talk about this in a follow on.